src/main/java/duke/parser/Parser.java

```diff
@@ -0,0 +1,326 @@
+package duke.parser;
+
+import duke.commands.*;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Todo;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.LocalTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+/**
+ * Parses user input.
+ */
+public class Parser {
+    // Input formats of date and times
+    private static final DateTimeFormatter dateTimeIn = DateTimeFormatter.ofPattern("yyyy-MM-dd HHmm");
+    private static final DateTimeFormatter dateIn = DateTimeFormatter.ofPattern("yyyy-MM-dd");
+    private static final DateTimeFormatter timeIn = DateTimeFormatter.ofPattern("HHmm");
```

Comment:
Constant should be uppercase and underline? I notice there are similar issues in other places.